controllers/nodefeaturediscovery_controller.go

```diff
@@ -37,29 +37,60 @@ import (
 	nfdv1 "github.com/kubernetes-sigs/node-feature-discovery-operator/api/v1"
 )
 
+// log is used to set the logger with a name that describes the actions of
+// functions and arguments in this file
 var log = logf.Log.WithName("controller_nodefeaturediscovery")
 
+// nfd is an NFD object that will be used to initialize the NFD operator
 var nfd NFD
 
 // NodeFeatureDiscoveryReconciler reconciles a NodeFeatureDiscovery object
 type NodeFeatureDiscoveryReconciler struct {
+
+	// client.Client reads and writes directly from/to the OCP API server. This
+	// field needs to be added to the reconciler because it is responsible for
+	// fetching objects from the server, which the NFD operator needs to do in
+	// order to add its labels to each node in the cluster.
 	client.Client
-	Log       logr.Logger
-	Scheme    *runtime.Scheme
-	Recorder  record.EventRecorder
+
+	// Log is used to log the reconciliation. Every controllers needs this.
+	Log logr.Logger
+
+	// Scheme is used by the kubebuilder library to set OwnerReferences. Every
+	// controller needs this.
+	Scheme *runtime.Scheme
+
+	// Recorder defines interfaces for working with OCP event recorders. This
+	// field is needed by NFD in order for NFD to write events.
+	Recorder record.EventRecorder
+
+	// AssetsDir defines the directory with assets under the operator image
 	AssetsDir string
 }
 
-// SetupWithManager sets up the controller with the Manager.
+// SetupWithManager sets up the controller with the Manager to create the controller.
+// The Manager serves the purpose of initializing shared dependencies (like caches
+// and clients) from the 'client.Client' field in the NodeFeatureDiscoveryReconciler
+// struct.
 func (r *NodeFeatureDiscoveryReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
-	// we want to initate reconcile loop only on spec change of the object
+	// The predicate package is used by the controller to filter events before
+	// they are sent to event handlers. In this case, the "validateUpdateEvent"
+	// function is used with UpdateFunc to initate the reconcile loop only on a
+	// spec change of the runtime object. (Refer to the "validateUpdateEvent"
+	// function in this file for more info.)
 	p := predicate.Funcs{
 		UpdateFunc: func(e event.UpdateEvent) bool {
 			return validateUpdateEvent(&e)
 		},
 	}
 
+	// To create a new controller, it must be managed by a manager that has
+	// already been created. The "For" function is used to to define the type of
+	// object being reconciled as "NFD". The "Owns" defines the types of objects
+	// being generated: DaemonSet, a Service, a ServiceAccount, some Pods, and
+	// some ConfigMaps. Finally, the "Complete" function builds the application
+	// controller using the NFD reconciler object.
```

Comment:
I think it's redundant (and possibly adds maintenance burden) to list all the object types in the comment as they're clearly visible from the code

```suggestion
	// Create a new controller.  "For" specifies the type of object being
	// reconciled whereas "Owns" specify the types of objects being
	// generated and "Complete" specifies the reconciler object.
```